AENS.md

```diff
@@ -66,19 +65,47 @@ could allow us an easier path for future update to the fee structures.
 
 Every entry in the `.test` namespace pays the same amount of fees.
 
+The new mechanism planted in Lima hard fork introduces auctions.
+
+The claim action becomes claim attempt. For short names claim transaction
+doesn't set ownership of the name. It can be followed by another
+claim from an account different than one set in preclaim for given name.
+Names below 32 characters are considered short.
+
+There is finite amount of time when the follow up claim is allowed.
+This time is expressed in height delta computed from function of length of the name.
+
+The function will return higher value for shorter names. In practice it means
+that the shorter the name, the longer another claim transaction is valid.
+Claim transaction becomes a bid in an auction.
+
+Furthermore, the initial fee for name is also a value of the function of
+a name length. The function is decreasing function: for shorter names the initial
+fee will be higher.
+
+Also bidding by claim transaction is constrained by price progression.
+Each next bid has to be higher by `X` tokens, determined by percent of the price
+defined in governance.
+
+All, functions, base fee, free length value and price progression may be subject
+to changes with governance mechanism. Non-bidding path of the name claim is purposed
+for development or testing.
+
 Each entry has a fixed expiration date on claim after which the entry should
 transition into the `revoked` state. Once it reaches this state, the name
 will be released, i.e. transition into the `unclaimed` state, after which
 it could be claimed again.
 
 To prevent the entry from expiring a user can, at any point before reaching
 the expiration date, update the entry, which pushes the expiration date
-into the future by the time delta of the registration time and the update
-time, e.g. if a user posts an update one week after claiming their name,
-the expiration date gets pushed one week further into the future.
-The main motivation of this expiration date is to prevent situations where
-the private keys, which control the name, are lost making the name unusable
-as well.
+into the future by at most the maximum rent time e.g. if a user posts an update
+one week after claiming their name, the expiration date gets pushed for a full
+term from the previous expiration date. Follow up updates will have no effect,
```

Comment:
So, I registered a name at `registration_height = 500` and we have a maximum rent time of `rent_blocks = 1000`, giving me `termination_height = 1500`. If I now post an update at `update_height = 750`, does that mean `termination_height := termination_height + rent_blocks = 2500 ` or `termination_height := termination_height + (update_height-registration_height) + rent_blocks = 1750`

Reply:
2500

The change seems to improve UX for business and similar use-cases. Risk of lose of the domain is smaller (easier to set alerts to pay the rent, when termination height is "less" random).

Also, we were thinking to change from ~100 days to annual rent (again, to improve UX).

What you think about it?

Reply:
I don't see how that is any less random than the current approach. How is it any better at avoiding the name squatting issue? Because you have to pay a year upfront instead of doing it in smaller increments?

With the old approach, I can basically always set a reminder to `termination_height - (rent_blocks/2)` and expand my name registration for another `rent_blocks` while still having an error margin of `rent_blocks/2`. With your approach, if I understand the »Also it limits rent time to one full next term« correctly, then instead of having a `rent_blocks/2` buffer, you could end up with a `rent_blocks*2` buffer? That's a 4x improvement as far as buffer time goes, but I don't see how that would benefit anyone in practical terms, when the rent time is ~1year.

Reply:
The `rent_blocks*2` is UX improvement.

I see this becomes controversial and I don't have big opinion here, so maybe we pull product stake holders here.

I understand, tho, the need to mimic more DNS, if we are up for popularizing the platform among regular business. People taking time-off, getting sick,... etc. eventually failing to re-new the name. If we have it happening in the same time of the year and we have months of backup it seems safer.

Current solution would work best, if it is offloaded to 3rd party for management or automated infrastructure.